wire/msgtx.go

```diff
@@ -880,19 +1140,27 @@ func (msg *MsgTx) LegacyBtcDecode(r io.Reader, pver uint32) error {
 	txOuts := make([]TxOut, count)
 	msg.TxOut = make([]*TxOut, count)
 	for i := uint64(0); i < count; i++ {
+		// The pointer is set now in case a script buffer is borrowed
+		// and needs to be returned to the pool on error.
 		to := &txOuts[i]
 		err = legacyReadTxOut(r, pver, msg.Version, to)
+		msg.TxOut[i] = to
 		if err != nil {
+			returnScriptBuffers()
 			return err
 		}
-		msg.TxOut[i] = to
+		totalScriptSize += uint64(len(to.PkScript))
 	}
 
 	msg.LockTime, err = binarySerializer.Uint32(r, littleEndian)
 	if err != nil {
+		returnScriptBuffers()
 		return err
 	}
 
+	writeTxInScriptsToMsgTx(msg, totalScriptSize)
```

Comment:
I think these need a comment to explain what's going on since this would be quite unintuitive for a reader that is not already familiar with this code.